src/interpreter/Interpreter.zig

```diff
@@ -0,0 +1,661 @@
+            return value;
+        },
+        else => {
+            LOG.warn("Unsupported node type: {s}", .{@tagName(node.kind)});
+
+            return self.reportError(
+                "I006",
+                "Unsupported node type: {s}",
+                .{@tagName(node.kind)},
+                error.UnsupportedNodeType,
+                .{
+                    .labels = &.{.{
+                        .color = .{ .basic = .red },
+                        .span = node.span.asReportz(),
+                        .message = "Unsupported node type.",
+                    }},
+                },
+            );
+        },
+    };
+}
+
+pub fn evalAssignment(self: *Self, tree: *const ast.Tree, node: ast.Node, env: *runtime.Environment, is_top_level: bool) !runtime.RuntimeValue {
+    const assignment = node.kind.assignment;
+    const name_node = tree.getNode(assignment.target) orelse return self.reportError(
+        "I010",
+        "Node with ID {} does not exist.",
+        .{assignment.target},
+        error.InvalidNodeId,
+        .{
+            .labels = &.{.{
+                .color = .{ .basic = .red },
+                .span = node.span.asReportz(),
+                .message = "Node ID out of bounds.",
+            }},
+        },
+    );
+
+    const name = switch (name_node.kind) {
+        .identifier => try self.getIdentifierName(tree, assignment.target),
+        .variable_ref => try self.getIdentifierName(tree, name_node.kind.variable_ref),
+        else => return error.UnsupportedNodeType,
+    };
+
+    const value = try self.evalNode(tree, assignment.value, env, is_top_level);
+
+    // Set the value in the environment.
+    switch (assignment.operator) {
+        .ASSIGN => try env.set(name, value),
+        // TODO: Implement other assignment operators.
+        else => return self.reportError(
+            "I003",
+            "Unsupported assignment operator: {s}",
+            .{@tagName(assignment.operator)},
+            error.UnsupportedNodeType,
+            .{
+                .labels = &.{.{
+                    .color = .{ .basic = .red },
+                    .span = node.span.asReportz(),
+                    .message = "Unsupported assignment operator.",
+                }},
+            },
+        ),
+    }
+
+    // Return the assigned value.
+    return value;
+}
+
+pub fn evalBinaryExpr(self: *Self, tree: *const ast.Tree, node: ast.Node, env: *runtime.Environment, is_top_level: bool) !runtime.RuntimeValue {
+    const binary_expr = node.kind.binary_operator;
+    const left_value = try self.evalNode(tree, binary_expr.left, env, is_top_level);
+    const right_value = try self.evalNode(tree, binary_expr.right, env, is_top_level);
+
+    return switch (binary_expr.operator) {
+        .ADD => blk: {
+            switch (left_value) {
+                .Integer => |li| switch (right_value) {
+                    .Integer => |ri| break :blk runtime.RuntimeValue{ .Integer = li + ri },
```

Comment:
It might be good to use decl literals instead of explicit `RuntimeValue` to improve readability and decrease length of the lines.
```suggestion
                    .Integer => |ri| break :blk .{ .Integer = li + ri },
```